gitpod-io/gitpod
# Polish handling of BillingMode

## Description
This PR includes a couple of alignments around BillingMode. It's reviewable by commit. As all changes share the same context I think it does not make sense to further split this up. :lotus_position: 

 1. Improved tests + additional tests (no change in behavior)
 2. Make "workspace classes" usable if either of these are true:
    1. You have the "workspace_classes" FF enabled
    2. The method `BillingMode.canSetWorkspaceClass` returns `true` for your BillingMode
 3. Small fix to BillingModes: Only have _paid_ UBP team memberships override existing Chargebee memberships
 4. Handle bad "CostCenter" error, in case you have not yet set a team spending limit for whatever reason
 5. ~Make `moment` time offset work on Usage List View~ removed
 6. Fix wrong usage of `findCustomerByUserId` and replace with `findCustomerByTeamId`

## Related Issue(s)

Fixes #12098
Fixes https://github.com/gitpod-io/gitpod/issues/12090
Fixes https://github.com/gitpod-io/gitpod/issues/12149

## How to test
 - [signup](https://gpl-ubp.preview.gitpod-dev.com/workspaces)
 - create a Team called `Gitpod-heroes` or so
 - check:
   - user preferences: no workspace classes selector, yet :heavy_check_mark: 
   - user billing: no Usage-based stuff, yet :heavy_check_mark: 
   - user settings menu: still has entries for Plans/Team Subscriptions :heavy_check_mark: 
 - try to start a workspace: shows the "set spending limit" UI :heavy_check_mark: 
 - signup for a Usage-Based team plan
   - user preferences: see workspace classes selector, yet :heavy_check_mark: 
   - user billing: see the UBP attribution selector :heavy_check_mark: 
   - user settings menu: no entries for Plans/Team Subscriptions :heavy_check_mark: 

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:
